### DISCUSSION TITLE
Wireguard

### DISCUSSION THREAD

**Discussion by harryv1** [RaspAP/raspap-webgui#1149] [General]
Hi, hopefully someone can help me. I am new on raspap and made an installation on a raspi4. The AP works fine but the wireguard implementation does not work. After entering the data manual in the GUI, the storage to the configuration file fails. What config files do I need and can I create and edit them manually? Tnx in advance.

**Comment by billz** [comment]
Have you followed the steps described in the docs? https://docs.raspap.com/wireguard/#manual-configuration

If you choose to manually configure WireGuard I'd strongly suggest you familiarize yourself with the protocol first, then create your config in the bash shell _before_ using RaspAP as a front-end. There are many tutorials online to assist you.

**Reply by harryv1** [reply]
I used the automated, script-based installation. I did not install openvpn but wireguard. Are there any dependencies? 
Trying to configure wireguard using the gui just generates a key-pair. But there are no files like wg0.conf.

**Reply by billz** [reply]
With a manual WG config you can create a server or peer configuration (or both). WG does not make a firm distinction between "servers" and "clients"... these terms are used for convenience only. Enable these options in **Settings > Enable server** and **Peer > Enable peer**,  generate public keys for the ones you want and choose **Save settings**. Depending on your choices, this will create the following files for you in `/etc/wireguard`:

```
client.conf
wg0.conf
wg-peer-private.key
wg-peer-public.key
wg-server-private.key
wg-server-public.key
```

These steps are illustrated in the video in the link shared above. 
No extra dependencies are required.
